filters/include/pcl/filters/impl/conditional_removal.hpp

```diff
@@ -694,31 +694,32 @@ pcl::ConditionalRemoval<PointT>::applyFilter (PointCloud &output)
 
   if (!keep_organized_)
   {
-    for (std::size_t cp = 0; cp < Filter<PointT>::indices_->size (); ++cp)
+    for (std::size_t index: (*Filter<PointT>::indices_))
```

Comment:
```suggestion
    for (const auto& index: (*Filter<PointT>::indices_))
```
PS: Is `Filter<PointT>::` qualifier needed?

Reply:
Yes, because there is no `using` for `indices_` here: https://github.com/PointCloudLibrary/pcl/blob/92ea8cf6b422466dbd382091d0643873c378d3fb/filters/include/pcl/filters/conditional_removal.h#L592-L597. Can be added, of course.

Reply:
> Can be added, of course

Seeing that `removed_indices_` and company are there, that might be better

Reply:
Pinging @facontidavide

Reply:
I have neutral opinion about this. I don't see any advantage not drawbacks.

Performance wise, it is irrelevant in this case, since the cost of getting the index by value instead of const reference is extremely low, if any.